pkg/api/types.go

```diff
@@ -335,6 +354,8 @@ type ProjectDirectoryImageBuildStepConfiguration struct {
 	// promoted unless explicitly targeted. Use for builds which
 	// are invoked only when testing certain parts of the repo.
 	Optional bool `json:"optional"`
+
+	*GitSourceReference `json:",inline"`
```

Comment:
By adding this and not replacing the current fields you are duplicating the API -- we should have this _instead_ of the current fields in this struct.

Reply:
So this should be removed, as it was in the previous commits?

Reply:
This is the last outstanding comment. Right now you have the `ContextDir` and the `DockerfilePath` in this configuration twice. You should remove the `ProjectDirectoryImageBuildStepConfiguration.ContextDir` and use only `ProjectDirectoryImageBuildStepConfiguration.GitSourceReference.ContextDir`. With the `json:",inline"` tag, the same JSON (or YAML) will get serialized correctly, since the embedded field uses the same config path.